File: HackerRank/CompareTheTriplets.rb
#!/bin/ruby

def solve(a0, a1, a2, b0, b1, b2)
    # Complete this function
    a = [a0,a1,a2];
    b = [b0,b1,b2];
    alice = 0;
    bob = 0;

    for i in 0..2 do
        if a[i] > b[i]; alice += 1; end
        if a[i] < b[i]; bob += 1; end
    end
    sol = [alice, bob];
end

a0,a1,a2 = gets.strip.split(' ')
a0 = a0.to_i
a1 = a1.to_i
a2 = a2.to_i
b0,b1,b2 = gets.strip.split(' ')
b0 = b0.to_i
b1 = b1.to_i
b2 = b2.to_i
result = solve(a0, a1, a2, b0, b1, b2)
print result.join(" ")

File: HackerRank/CompareTheTriplets.cpp
#include <bits/stdc++.h>

using namespace std;

vector < int > solve(int a0, int a1, int a2, int b0, int b1, int b2){
    // Complete this function
    vector<int> points(2,0);
  
    if (a0 > b0) points[0]++; else 
    if (a0 < b0) points[1]++;
  
    if (a1 > b1) points[0]++; else 
    if (a1 < b1) points[1]++;
  
    if (a2 > b2) points[0]++; else
    if (a2 < b2) points[1]++;
  
    return points;
}

int main() {
    int a0;
    int a1;
    int a2;
    cin >> a0 >> a1 >> a2;
    int b0;
    int b1;
    int b2;
    cin >> b0 >> b1 >> b2;
    vector < int > result = solve(a0, a1, a2, b0, b1, b2);
    for (ssize_t i = 0; i < result.size(); i++) {
        cout << result[i] << (i != result.size() - 1 ? " " : "");
    }
    cout << endl;
    

    return 0;
}
